pkg/client/client.go

```diff
@@ -139,3 +167,103 @@ func (c *Client) GetInstances(serviceType string) (map[string]string, error) {
 	}
 	return instances, nil
 }
+
+func (c *Client) CreateServiceInstance(instanceName, serviceName, servicePlan, resourceGrp, region string) (string, error) {
+	//Find ServiceName to be created
+	service, err := c.ResCatalogAPI.FindByName(serviceName, true)
+	if err != nil {
+		return "", err
+	}
+
+	servicePlanID, err := c.ResCatalogAPI.GetServicePlanID(service[0], servicePlan)
+	if err != nil {
+		return "", err
+	}
+
+	if servicePlanID == "" {
+		_, err := c.ResCatalogAPI.GetServicePlan(servicePlan)
+		if err != nil {
+			return "", err
+		}
+		servicePlanID = servicePlan
+	}
+
+	deployments, err := c.ResCatalogAPI.ListDeployments(servicePlanID)
+	if err != nil {
+		return "", err
+	}
+
+	if len(deployments) == 0 {
+		klog.Infof("No deployment found for service plan : %s", servicePlan)
+		return "", err
+	}
+
+	supportedDeployments := []models.ServiceDeployment{}
+	supportedLocations := make(map[string]bool)
+	for _, d := range deployments {
+		if d.Metadata.RCCompatible {
+			deploymentLocation := d.Metadata.Deployment.Location
+			supportedLocations[deploymentLocation] = true
+			if deploymentLocation == region {
+				supportedDeployments = append(supportedDeployments, d)
+			}
+		}
+	}
+
+	if len(supportedDeployments) == 0 {
+		locationList := make([]string, 0, len(supportedLocations))
+		for l := range supportedLocations {
+			locationList = append(locationList, l)
+		}
+		return "", fmt.Errorf("No deployment found for service plan %s at location %s.\nValid location(s) are: %q.\nUse service instance example if the service is a Cloud Foundry service.",
+			servicePlan, region, locationList)
+	}
+
+	var resourceGroup models.ResourceGroup
+	if resourceGrp == "" {
+		resourceGroupQuery := management.ResourceGroupQuery{
+			Default: true,
+		}
+
+		grpList, err := c.ResGroupAPI.List(&resourceGroupQuery)
+		if err != nil {
+			return "", err
+		}
+		resourceGroup = grpList[0]
+
+	} else {
+		grp, err := c.ResGroupAPI.FindByName(nil, resourceGrp)
+		if err != nil {
+			return "", err
+		}
+		resourceGroup = grp[0]
+	}
+
+	klog.Infof("Resource group: %s and ID: %s", resourceGroup.Name, resourceGroup.ID)
+
+	var serviceInstancePayload = controller.CreateServiceInstanceRequest{
+		Name:            instanceName,
+		ServicePlanID:   servicePlanID,
+		ResourceGroupID: resourceGroup.ID,
+		TargetCrn:       supportedDeployments[0].CatalogCRN,
+	}
+
+	serviceInstance, err := c.ResourceClientV1.CreateInstance(serviceInstancePayload)
+	if err != nil {
+		return "", err
+	}
+
+	klog.Infof("Resource service Instance Details :%v\n", serviceInstance)
+	klog.Infof("Resource service InstanceID :%v\n", serviceInstance.Crn.ServiceInstance)
+
+	return serviceInstance.Crn.ServiceInstance, nil
+}
+
+func (c *Client) DeleteServiceInstance(instanceID string, recursive bool) error {
+	err := c.ResourceClientV1.DeleteInstance(instanceID, recursive)
+	if err != nil {
+		klog.Infof("Failed to delte the instance with id %s because of the error %s", instanceID, err)
```

Comment:
fix the typo
```suggestion
		klog.Infof("Failed to delete the instance with id %s because of the error %s", instanceID, err)
```

and this can be a warning instead of info